File: backend/app/schemas/health.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class HealthDependency(BaseModel):
    name: str
    status: Literal["healthy", "degraded", "unavailable"]
    required: bool
    details: dict[str, str] = Field(default_factory=dict)


class HealthData(BaseModel):
    status: Literal["healthy", "degraded", "unavailable"]
    service: str
    environment: str
    version: str
    request_id: str | None = None
    dependencies: list[HealthDependency] = Field(default_factory=list)

File: frontend/src/services/health.ts
import { env } from 'lib/env';

export type HealthDependency = {
  name: string;
  status: 'healthy' | 'degraded' | 'unavailable';
  required: boolean;
  details: Record<string, string>;
};

export type HealthResponse = {
  success: boolean;
  data: {
    status: 'healthy' | 'degraded' | 'unavailable';
    service: string;
    environment: string;
    version: string;
    dependencies: HealthDependency[];
  };
  error: null;
  meta: Record<string, unknown>;
};

export async function fetchHealthDetails(): Promise<HealthResponse> {
  const response = await fetch(`${env.apiBaseUrl}/health/details`);
  if (!response.ok) {
    throw new Error('Failed to load platform health details.');
  }
  return (await response.json()) as HealthResponse;
}
